src/main/java/com/hubspot/jinjava/interpret/Context.java

```diff
@@ -385,6 +388,18 @@ public void setRenderDepth(int renderDepth) {
     this.renderDepth = renderDepth;
   }
 
+  public void pushRenderStack(String template) {
+    renderStack.push(template);
+  }
+
+  public String popRenderStack() {
+    return renderStack.pop();
+  }
+
+  public Stack getRenderStack() {
```

Comment:
If we really want to protect the underlying stack from being modified outside the Context, then I think you just need to implement `doesStackContain()`. Otherwise this allows total access to the stack contents.